CHANGELOG.md

```diff
@@ -1,5 +1,663 @@
+   peers). We had to do this because we had to assume that _most_ peers didn't
+   have the requested block.
+
+#### Pubsub
+
+This release includes some significant reliability improvements in pubsub
+subscription handling. If you've previously had issues with connected pubsub
+peers _not_ seeing each-other's messages, please upgrade ASAP.
+
+#### Reuseport
+
+In this release, we've rewritten our previously error-prone `go-reuseport`
+library to _not_ duplicate a significant portion of Go's low-level networking
+code. This was made possible by Go's new `Control`
+[`net.Dialer`](https://golang.org/pkg/net/#Dialer) option.
+
+In the past, our first suggestion to anyone experiencing weird resource or
+connectivity issues was to disable `REUSEPORT` (set `IPFS_REUSEPORT` to false).
+This should no longer be necessary.
+
+#### Badger Datastore
+
+[Badger has reached 1.0][badger-release]. This release brings an audit and
+numerous reliability fixes. We are now reasonably confident that badger will
+become the default datastore in a future release.
+
+[badger-release]: https://blog.dgraph.io/post/releasing-v1.0/
+
+This release also adds a new `Truncate` configuration option for the badger
+datastore (enabled by default for new IPFS nodes). When enabled, badger will
+_delete_ any un-synced data on start instead of simply refusing to start. This
+should be safe on all filesystems where the `sync` operation is safe and removes
+the need for manual intervention when restarting an IPFS node after a crash.
+
+### Refactors and Endeavors
+
+#### Commands Library
+
+The legacy commands library shim has now been completely removed. This won't
+mean much for many users but the go-ipfs team is happy to have this behind them.
+
+#### Base32 CIDs
+
+This release can now encode CIDs in responses in bases other than base58. This
+is primarily useful for web-browser integration as it allows us to (a) encode
+CIDs in a lower-case base (e.g., base32) and then use them in the _origin_ part
+of URLs. The take away is: this release brings us a step closer to better
+browser integration.
+
+Specifically, this release adds two flags:
+
+1. `--cid-base`: When specified, the IPFS CLI will encode all CIDv1 CIDs using the
+   requested base.
+2. `--upgrade-cidv0-in-output`: When specified, the IPFS CLI will _upgrade_ CIDv0
+   CIDs to CIDv1 CIDs when returning them to the user. This upgrade is necessary
+   because CIDv0 doesn't support multibase however, it's off by default as it
+   changes the _binary_ representation of the CIDs (which could have unintended
+   consequences).
+
+#### CoreAPI
+
+The work on the CoreAPI refactor ([ipfs/go-ipfs#4498][]) has progressed leaps and
+bounds this release. The CoreAPI is a comprehensive programmatic interface
+designed to allow go-ipfs be used as a daemon or a library interchangeably.
+
+As of this release, go-ipfs now has:
+
+* External interface definitions in [ipfs/interface-go-ipfs-core][].
```

Comment:
these are all missing contents

Reply:
That's short-hand for `[ipfs/interface-go-ipfs-core][ipfs/interface-go-ipfs-core]` (which refers to the link definition `[ipfs/interface-go-ipfs-core]: https://github.com/ipfs/interface-go-ipfs-core` at the end of the section.